src/Test/Utilities/Portable/Compilation/ControlFlowGraphVerifier.cs

```diff
@@ -766,8 +766,9 @@ private static bool CanBeInControlFlowGraph(IOperation n)
                              !ITypeSymbolHelpers.IsDynamicType(binary.Type));
 
                 case OperationKind.InstanceReference:
-                    // Implicit instance receivers are expected to have been removed when dealing with creations.
-                    return ((IInstanceReferenceOperation)n).ReferenceKind == InstanceReferenceKind.ContainingTypeInstance;
+                    // Implicit instance receivers, except for anonymous type creations, are expected to have been removed when dealing with creations.
+                    return ((IInstanceReferenceOperation)n).ReferenceKind == InstanceReferenceKind.ContainingTypeInstance ||
+                        ((IInstanceReferenceOperation)n).ReferenceKind == InstanceReferenceKind.ImplicitReceiver && n.Type.IsAnonymousType;
```

Comment:
>((IInstanceReferenceOperation)n).ReferenceKind == InstanceReferenceKind.ImplicitReceiver && n.Type.IsAnonymousType [](start = 24, length = 114)

I think we should strengthen this condition by checking that the parent is an anonymous type property access, its parent is an assignment and its parent is an anonymous object creation.